src/Initializer/InitProcedure/InitMesh.cpp

```diff
@@ -130,6 +131,97 @@ size_t getNumOutgoingEdges(seissol::geometry::MeshReader& meshReader) {
+  unsigned int cubeMinY = boundaries.cubeMinY;
+  unsigned int cubeMaxY = boundaries.cubeMaxY;
+  unsigned int cubeMinZ = boundaries.cubeMinZ;
+  unsigned int cubeMaxZ = boundaries.cubeMaxZ;
+
+  unsigned int cubeX = dims.cubeX;
+  unsigned int cubeY = dims.cubeY;
+  unsigned int cubeZ = dims.cubeZ;
+
+  unsigned int cubePx = partitions.cubePx;
+  unsigned int cubePy = partitions.cubePy;
+  unsigned int cubePz = partitions.cubePz;
+
+  double cubeScale = scaling.cubeS;
+  double cubeScaleX = scaling.cubeSx;
+  double cubeScaleY = scaling.cubeSy;
+  double cubeScaleZ = scaling.cubeSz;
+
+  double cubeTx = translation.cubeTx;
+  double cubeTy = translation.cubeTy;
+  double cubeTz = translation.cubeTz;
+ 
+  // create additional variables necessary for cubeGenerator()
+  unsigned int numCubes[4];
+  unsigned int numPartitions[4];
+
+  // check input arguments
+  numCubes[0] = cubeX;
+  numCubes[1] = cubeY;
+  numCubes[2] = cubeZ;
+
+  numPartitions[0] = cubePx;
+  numPartitions[1] = cubePy;
+  numPartitions[2] = cubePz;
+
+  for (int i = 0; i < 3; i++) {
+    if (numCubes[i] < 2)
+      logError() << "Number of cubes in" << dim2str(i) << "dimension must be at least 2";
+      if (numCubes[i] % numPartitions[i] != 0)
+        logError() << "Number of cubes in" << dim2str(i)
+                   << "dimension can not be distribute to" << numPartitions[i] << "partitions";
+      if ((numCubes[i] / numPartitions[i]) % 2 != 0)
+        logError() << "Number of cubes per partition in" << dim2str(i) << "dimension must be a multiple of 2";
+  }
+
+  // Compute additional sizes
+  numCubes[3] = numCubes[0] * numCubes[1] * numCubes[2];
+  numPartitions[3] = numPartitions[0] * numPartitions[1] * numPartitions[2];
+
+  unsigned int numCubesPerPart[4];
+  unsigned long numElemPerPart[4];
+  for (int i = 0; i < 4; i++) {
+    numCubesPerPart[i] = numCubes[i] / numPartitions[i];
+    numElemPerPart[i] = numCubesPerPart[i] * 5;
+  }
+
+  unsigned int numVrtxPerPart[4];
+  for (int i = 0; i < 3; i++)
+          numVrtxPerPart[i] = numCubesPerPart[i] + 1;
+  numVrtxPerPart[3] = numVrtxPerPart[0] * numVrtxPerPart[1] * numVrtxPerPart[2];
+
+  unsigned int numBndElements[3];
+  numBndElements[0] = 2*numCubesPerPart[1]*numCubesPerPart[2];
+  numBndElements[1] = 2*numCubesPerPart[0]*numCubesPerPart[2];
+  numBndElements[2] = 2*numCubesPerPart[0]*numCubesPerPart[1];
+
+  // output file name
+  auto cubeOutput = seissolParams.mesh.meshFileName + ".nc";  // mesh file name doesnt include file type ending
+  std::string fileName = seissolParams.mesh.meshFileName + ".nc";
+
+  // TODO: I think passing output into the cubeGenerator function breaks the call? there was a case where the output name's string representation was suddenly empty when passing output to the funciton
+  // instead, replace the outut variable with its value in the function call
+  logInfo() << "Start generating a mesh using the CubeGenerator";
+  cubeGenerator(numCubes, numPartitions, \
```

Comment:
Send a `struct` or `class` as parameter here (instead of passing all arguments individually)